File: Python/Maximum Depth of Binary Tree/Maximum Depth.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        return self.Helper(root, root, 1)

    def Helper(self, left, right, depth):
        if (not left and not right):
            return depth - 1
        return max(
            self.Helper(left.left, left.right, depth + 1) if left else depth,
            self.Helper(right.right, right.left, depth + 1)
            if right else depth)

File: C#/Maximum Depth of Binary Tree/MaximumDepth.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int MaxDepth (TreeNode root) {
        if (root == null) return 0;
        return Helper (root, 1);
    }

    private int Helper (TreeNode node, int depth) {
        if (node == null) return --depth;
        int left = Helper (node.left, ++depth);
        int right = Helper (node.right, depth);
        return left > right?left : right;
    }
}